presto-spi/src/main/java/com/facebook/presto/spi/block/MapBlockBuilder.java

```diff
@@ -180,6 +181,11 @@ public SingleMapBlockWriter beginBlockEntry()
 
     @Override
     public BlockBuilder closeEntry()
+    {
+        return closeEntry(Optional.empty());
+    }
+
+    private BlockBuilder closeEntry(Optional<ProvidedHashTable> providedHashTable)
```

Comment:
I'm a little concerned about the overhead of object creation here: `Optional` and `ProvidedHashTable`. I'm ok with its current shape. I just want to make sure you've thought about it.